kernel/kselftest.py

```diff
@@ -80,10 +80,20 @@ def setUp(self):
         if self.run_type == 'upstream':
             location = self.params.get('location', default='https://github.c'
                                        'om/torvalds/linux/archive/master.zip')
-            tarball = self.fetch_asset("kselftest.zip", locations=[location],
-                                       expire='1d')
-            archive.extract(tarball, self.workdir)
-            self.buldir = os.path.join(self.workdir, 'linux-master')
+            path = ''
+            if '.zip' in location:
+                tarball = self.fetch_asset("kselftest.zip", locations=[location],
+                                           expire='1d')
+                archive.extract(tarball, self.workdir)
+                path = glob.glob(os.path.join(self.workdir, "linux*"))
+
+            else:
+                git.get_repo(location, destination_dir=self.workdir)
+                path = glob.glob(self.workdir)
```

Comment:
This will return [`self.workdir`] , so not sure how the next steps would work since the workdir won't have a `Makefile`.

Reply:
no, i think git clone in self.workdir as first in initialize an empty git than get content from given git location and put in self.workdir so it should has all directory present

Reply:
for better understanding below log will help 
```
2020-07-29 06:22:29,796 git              L0092 DEBUG| Initializing new git repo at /var/tmp/avocado__ovku1vx/avocado_job_5mt0zeag/1-avocado-misc-tests_kernel_kselftest.py_kselftest.test_power-upstream-2fd3 for receiving git repo http://ltc-jenkins.aus.stglabs.ibm.com:81/gits/linux-p10/
2020-07-29 06:22:29,796 process          L0607 INFO | Running '/usr/bin/git init'
2020-07-29 06:22:29,814 process          L0439 DEBUG| [stdout] Initialized empty Git repository in /var/tmp/avocado__ovku1vx/avocado_job_5mt0zeag/1-avocado-misc-tests_kernel_kselftest.py_kselftest.test_power-upstream-2fd3/.git/
2020-07-29 06:22:29,814 process          L0689 INFO | Command '/usr/bin/git init' finished with 0 after 0.01616215705871582s
2020-07-29 06:22:29,814 git              L0114 INFO | Fetching git [REP 'http://ltc-jenkins.aus.stglabs.ibm.com:81/gits/linux-p10/' BRANCH 'master'] -> /var/tmp/avocado__ovku1vx/avocado_job_5mt0zeag/1-avocado-misc-tests_kernel_kselftest.py_kselftest.test_power-upstream-2fd3
2020-07-29 06:22:29,814 process          L0607 INFO | Running '/usr/bin/git fetch -q -f -u -t http://ltc-jenkins.aus.stglabs.ibm.com:81/gits/linux-p10/ master:master'
2020-07-29 06:30:59,586 process          L0689 INFO | Command '/usr/bin/git fetch -q -f -u -t http://ltc-jenkins.aus.stglabs.ibm.com:81/gits/linux-p10/ master:master' finished with 0 after 509.76993012428284s
2020-07-29 06:30:59,586 git              L0147 DEBUG| Checking out branch master
2020-07-29 06:30:59,586 process          L0607 INFO | Running '/usr/bin/git checkout master'
2020-07-29 06:31:09,970 process          L0439 DEBUG| [stderr] Already on 'master'
2020-07-29 06:31:09,973 process          L0689 INFO | Command '/usr/bin/git checkout master' finished with 0 after 10.385160207748413s
2020-07-29 06:31:09,973 git              L0157 DEBUG| Specific commit not specified
2020-07-29 06:31:09,973 process          L0607 INFO | Running '/usr/bin/git log --pretty=format:%H -1'
2020-07-29 06:31:09,976 process          L0448 DEBUG| [stdout] fd67c6c03c804bce1c546ad774b94e4e29549657
2020-07-29 06:31:09,976 process          L0689 INFO | Command '/usr/bin/git log --pretty=format:%H -1' finished with 0 after 0.0015225410461425781s
2020-07-29 06:31:09,977 process          L0607 INFO | Running '/usr/bin/git describe'
2020-07-29 06:31:09,979 process          L0439 DEBUG| [stderr] fatal: No names found, cannot describe anything.
2020-07-29 06:31:09,979 process          L0689 INFO | Command '/usr/bin/git describe' finished with 128 after 0.0012977123260498047s
2020-07-29 06:31:09,980 git              L0165 INFO | git commit ID is b'fd67c6c03c804bce1c546ad774b94e4e29549657' (no tag found)
2020-07-29 06:31:09,980 output           L0660 DEBUG| [stdout] ['/var/tmp/avocado__ovku1vx/avocado_job_5mt0zeag/1-avocado-misc-tests_kernel_kselftest.py_kselftest.test_power-upstream-2fd3']
```
self.workdir content  so that we are processing in next setup 
```
[root@system ~]# ll /var/tmp/avocado__ovku1vx/avocado_job_5mt0zeag/1-avocado-misc-tests_kernel_kselftest.py_kselftest.test_power-upstream-2fd3
total 824
drwxr-xr-x.  27 root root   4096 Jul 29 06:31 arch
drwxr-xr-x.   3 root root   4096 Jul 29 06:31 block
drwxr-xr-x.   2 root root   4096 Jul 29 06:31 certs
-rw-r--r--.   1 root root    496 Jul 29 06:30 COPYING
-rw-r--r--.   1 root root  99752 Jul 29 06:30 CREDITS
drwxr-xr-x.   4 root root   4096 Jul 29 06:31 crypto
drwxr-xr-x.  79 root root   4096 Jul 29 06:31 Documentation
drwxr-xr-x. 140 root root   4096 Jul 29 06:31 drivers
drwxr-xr-x.  79 root root   4096 Jul 29 06:31 fs
drwxr-xr-x.  29 root root   4096 Jul 29 06:31 include
drwxr-xr-x.   2 root root   4096 Jul 29 06:31 init
drwxr-xr-x.   2 root root   4096 Jul 29 06:31 ipc
-rw-r--r--.   1 root root   1327 Jul 29 06:31 Kbuild
-rw-r--r--.   1 root root    555 Jul 29 06:31 Kconfig
drwxr-xr-x.  19 root root   4096 Jul 29 06:31 kernel
drwxr-xr-x.  20 root root  12288 Jul 29 06:31 lib
drwxr-xr-x.   6 root root   4096 Jul 29 06:31 LICENSES
-rw-r--r--.   1 root root 561058 Jul 29 06:31 MAINTAINERS
-rw-r--r--.   1 root root  61873 Jul 29 06:31 Makefile
drwxr-xr-x.   3 root root   4096 Jul 29 06:31 mm
drwxr-xr-x.  72 root root   4096 Jul 29 06:31 net
-rw-r--r--.   1 root root    727 Jul 29 06:31 README
drwxr-xr-x.  31 root root   4096 Jul 29 06:31 samples
drwxr-xr-x.  16 root root   4096 Jul 29 06:31 scripts
drwxr-xr-x.  13 root root   4096 Jul 29 06:31 security
drwxr-xr-x.  26 root root   4096 Jul 29 06:31 sound
drwxr-xr-x.  37 root root   4096 Jul 29 06:31 tools
drwxr-xr-x.   3 root root   4096 Jul 29 06:31 usr
drwxr-xr-x.   4 root root   4096 Jul 29 06:31 virt
[root@system ~]# 
```

Reply:
got it, thanks.